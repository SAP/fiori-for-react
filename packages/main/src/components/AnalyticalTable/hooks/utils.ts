export const makeTemplateColumns = (columns, resizedColumns) =>
  columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
