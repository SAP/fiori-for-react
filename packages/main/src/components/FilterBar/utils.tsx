import { cloneElement, ReactElement } from 'react';

export const setPropsOfChildren = (children, currentRef) => {
  return children.map((child) => {
    if (child.props.children.props[currentRef]?.children) {
      const childrenOfChild = child.props.children;
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SELECT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-MULTI-COMBOBOX'
      ) {
        const selectedIndices = Array.from(child.props.children.props[currentRef].children)
          .map((item, index) => (item._state.selected ? index : false))
          .filter((el) => el !== false);
        const options = child.props.children.props.children.map((item, index) => {
          if (selectedIndices.includes(index)) {
            return cloneElement(item, { selected: true });
          }
          return cloneElement(item, { selected: false });
        });
        return cloneElement(child, {
          children: { ...childrenOfChild, props: { ...childrenOfChild.props, children: options } }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-INPUT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-DATEPICKER'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, value: child.props.children.props[currentRef]._state.value }
          }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SWITCH' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-CHECKBOX'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, checked: child.props.children.props[currentRef]._state.checked }
          }
        });
      }
      return child;
    }
    return child;
  });
};

export const addRef = (children, refsArray, currentRef) =>
  children.map((child) => {
    const childRef = refsArray.filter((item) => item.key === child.key);
    const childrenAttributes = child.props.children;
    const childrenProps = child.props.children.props;
    return cloneElement(child as ReactElement<any>, {
      children: { ...childrenAttributes, props: { ...childrenProps, [currentRef]: childRef[0]?.node } }
    });
  });

export const renderSearchWithValue = (searchElement, searchValue) => {
  return cloneElement(searchElement(), { value: searchValue });
};
