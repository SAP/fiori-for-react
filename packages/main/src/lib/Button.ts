import { Button, ButtonPropTypes } from '../webComponents/Button';

export { Button, ButtonPropTypes };
