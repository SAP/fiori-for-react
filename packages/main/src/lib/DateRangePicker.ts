import { DateRangePicker } from '../webComponents/DateRangePicker';

export { DateRangePicker };
