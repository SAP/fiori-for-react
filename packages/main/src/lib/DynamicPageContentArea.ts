import { DynamicPageContentArea } from '../components/DynamicPageContentArea';

export { DynamicPageContentArea };
