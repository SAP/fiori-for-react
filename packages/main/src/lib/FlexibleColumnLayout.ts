import { FlexibleColumnLayout } from '../webComponents/FlexibleColumnLayout';

export { FlexibleColumnLayout };
