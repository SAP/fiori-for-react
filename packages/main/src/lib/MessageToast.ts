import { MessageToast } from '../components/MessageToast';

export { MessageToast };

// Required for React lazy
export default MessageToast;
