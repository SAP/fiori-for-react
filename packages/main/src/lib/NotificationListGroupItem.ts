import { NotificationListGroupItem } from '../webComponents/NotificationListGroupItem';

export { NotificationListGroupItem };
