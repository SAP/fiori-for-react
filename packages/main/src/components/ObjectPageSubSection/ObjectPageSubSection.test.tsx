import { createPassThroughPropsTest } from '@shared/tests/utils';
import { ObjectPageSubSection } from '@ui5/webcomponents-react/lib/ObjectPageSubSection';
import { render } from '@shared/tests';
import React from 'react';

describe('ObjectPageSubSection', () => {
  test('Render without Crashing', () => {
    // @ts-ignore
    const { asFragment } = render(<ObjectPageSubSection id="test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('No ID should throw', () => {
    // @ts-ignore
    const renderer = () => shallow(<ObjectPageSubSection.InnerComponent title="test" classes={{}} />);
    expect(renderer).toThrow();
  });

  createPassThroughPropsTest(ObjectPageSubSection);
});
