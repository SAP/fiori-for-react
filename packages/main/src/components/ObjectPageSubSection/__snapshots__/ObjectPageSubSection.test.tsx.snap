// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectPageSubSection Render without Crashing 1`] = `"<div class=\\"ObjectPageSubSection-objectPageSubSectionHeaderTitle-0-2-2\\"></div><div class=\\"ObjectPageSubSection-subSectionContent-0-2-3\\"></div>"`;
